src/components/drawing/index.js

```diff
@@ -112,6 +112,7 @@ drawing.lineGroupStyle = function(s, lw, lc, ld) {
 };
 
 drawing.dashLine = function(s, dash, lineWidth) {
+    lineWidth = +lineWidth || 0;
```

Comment:
I think this does fix #166 - and https://github.com/plotly/plotly.js/pull/1332/commits/4aa1a4535a63633f38256159fc751ca59dd7b2bd should catch any of these `nanpx` or `undefinedpx` errors, at least as long as they go through d3.

Will investigate the other two tomorrow, they may be related.